sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -343,4 +354,144 @@ export class AssessmentSequentialProcessingService {
       options?.alternativeReferenceDate,
     );
   }
+
+  /**
+   * Get SFAS application part time awards data from either the SFAS application table or
+   * the SFAS part time applications table depending on the SFAS application query type.
+   * @param assessmentId Current assessment being processed.
+   * @param sfasApplicationType SFAS application query type (sfasApplication or sfasPTApplication).
+   * @returns SFAS application part time awards data.
+   */
+  async getProgramYearLegacyAwardsTotals(
+    assessmentId: number,
+    sfasApplicationType: string,
+  ): Promise<SFASApplicationAwards> {
+    const assessment = await this.studentAssessmentRepo.findOne({
+      select: {
+        id: true,
+        application: {
+          id: true,
+          applicationNumber: true,
+          student: {
+            id: true,
+            birthDate: true,
+            sinValidation: {
+              sin: true,
+            },
+            user: {
+              lastName: true,
+            },
+          },
+          programYear: {
+            id: true,
+            startDate: true,
+          },
+        },
+        assessmentDate: true,
+      },
+      relations: {
+        application: {
+          student: { sinValidation: true, user: true },
+          programYear: true,
+        },
+      },
+      where: {
+        id: assessmentId,
+      },
+    });
+    if (!assessment) {
+      throw new Error(`Assessment if ${assessmentId} not found.`);
+    }
+    const lastName = assessment.application.student.user.lastName;
+    const sin = assessment.application.student.sinValidation.sin;
+    const birthDate = assessment.application.student.birthDate;
+    const programYearStartDate = assessment.application.programYear.startDate;
+    // Get the first assessment date for the current application. If there are multiple
+    // assessments for the current application, then set the first assessment date to
+    // the assessment date of the first assessment.
+    let firstAssessmentDate = getISODateOnlyString(assessment.assessmentDate);
+    const firstAssessment = await this.dataSource.manager
+      .getRepository(StudentAssessment)
+      .createQueryBuilder("assessment")
+      .select("assessment.assessmentDate")
+      .innerJoin("assessment.application", "application")
+      .where("assessment.id = :assessmentId", { assessmentId: assessment.id })
+      .where("application.applicationNumber = :applicationNumber", {
+        applicationNumber: assessment.application.applicationNumber,
+      })
+      .orderBy("assessment.assessmentDate")
+      .getOne();
+    if (firstAssessment) {
+      firstAssessmentDate = getISODateOnlyString(
+        firstAssessment.assessmentDate,
+      );
+    }
```

Comment:
We do not need this query to get the first assessment date. Also keep in mind, during the calculation of original assessment `assessment_date` of the assessment will be null until all the calculations are complete.
We have already implemented the logic to find the reference assessment date from `getSequencedApplications()` method.


We can get the reference assessment date from the `getSequencedApplications()`  from the `SequencedApplications` object. 

It is calculated here. But needs to be set in a variable to be returned.

![image](https://github.com/bcgov/SIMS/assets/54600590/1a65b310-2b1c-457f-a216-948313283a84)